RealmSwift/Optional.swift

```diff
@@ -32,23 +32,12 @@ extension Double: RealmOptionalType {}
 extension Bool: RealmOptionalType {}
 
 // Not all RealmOptionalType's can be cast to AnyObject, so handle casting logic here.
-private func realmOptionalToAnyObject<T: RealmOptionalType>(value: T?) -> AnyObject? {
-    if let anyObjectValue: AnyObject = value as? AnyObject {
-        return anyObjectValue
-    } else if let int8Value = value as? Int8 {
-        return NSNumber(value: int8Value)
-    } else if let int16Value = value as? Int16 {
-        return NSNumber(value: int16Value)
-    } else if let int32Value = value as? Int32 {
-        return NSNumber(value: int32Value)
-    } else if let int64Value = value as? Int64 {
-        return NSNumber(value: int64Value)
-    }
-    return nil
+private func realmOptionalToAnyObject<T: RealmOptionalType>(value: T?) -> Any? {
```

Comment:
I believe this will cause incorrect behavior since `Int8`/`Int16`/`Int32`/`Int64` are not (yet) bridged to `NSNumber`. Instead, they're boxed in a Swift value container which will probably crash the existing Objective-C code.